src/components/category-form/category-form.js

```diff
@@ -0,0 +1,249 @@
+      .required(CATEGORY_ERROR_MESSAGE),
+    uaName: Yup.string()
+      .min(2, CATEGORY_VALIDATION_ERROR)
+      .required(CATEGORY_ERROR_MESSAGE),
+      code: Yup.string()
+      .min(2, CATEGORY_VALIDATION_ERROR)
+      .required(CATEGORY_ERROR_MESSAGE)
+  });
+
+  const {
+    values,
+    handleSubmit,
+    handleChange,
+    touched,
+    errors,
+    setFieldValue
+  } = useFormik({
+    validationSchema: categoryValidationSchema,
+    initialValues: {
+      categoryImage: category.images.thumbnail || '',
+      uaName: category.name[0].value || '',
+      enName:category.name[1].value || '',
+      code:category.code || ''
+    },
+    onSubmit: (data) => {
+      const newCategory = createCategory(data);
+       if (edit) {
+        dispatch(updateCategory({ id,category: newCategory,upload }));
+        return;
+      }
+      dispatch(addCategory({category: newCategory,upload }));
+     
+    }
+  });
+
+  const handleImageLoad = (e) => {
+    if (e.target.files && e.target.files[0]) {
+      const reader = new FileReader();
+      reader.onload = (event) => {
+        setFieldValue('categoryImage', event.target.result);
+        setCategoryImage(event.target.result);
+      };
+      reader.readAsDataURL(e.target.files[0]);
+      setUpload(e.target.files[0]);
+    }
+  };
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <Grid item xs={12}>
+          <Paper className={styles.categoryItemUpdate}>
+            <span className={styles.imageUpload}>
+              {config.labels.avatarText}
+            </span>
+            <ImageUploadContainer
+              handler={handleImageLoad}
+              srcForAvatar={
+                categoryImage ||
+                `${config.imagePrefix}${values.categoryImage}`
+              }
+                fileName={upload.name ||category.images.thumbnail}
+            />
+            <TextField
+              data-cy='code'
+              name='code'
+              className={styles.textField}
+              variant='outlined'
+              label={'Код категорії'}
+              value={values.code}
+              onChange={handleChange}
+              error={touched.code && !!errors.code}
+            />
+            {touched.code && errors.code && (
+              <div data-cy='code-error' className={styles.inputError}>
+                {errors.code}
+              </div>
+            )}
+          </Paper>
+        </Grid>
+        <AppBar position='static'>
+          <Tabs
+            className={styles.tabs}
+            value={tabsValue}
+            onChange={handleTabsChange}
+            aria-label='simple tabs example'
+          >
+            {languageTabs}
+          </Tabs>
+        </AppBar>
+        {languages.map((lang, index) => (
+          <TabPanel key={index} value={tabsValue} index={index}>
+            <Paper className={styles.categoryItemUpdate}>
+               <TextField
+                data-cy={`${lang}Name`}
```

Comment:
Why are you using camel case here. It's better to set kebab case.

Reply:
As far as I know we use kebab case only with naming of our folders and files

Reply:
In cypress tests to pick element we are using data-cy attribute, that named by kebab case.

Reply:
![image](https://user-images.githubusercontent.com/59802190/101043274-cf9d1480-3586-11eb-835c-864806fe4c38.png)
For example category cypress tests.

Reply:
I have all ready change